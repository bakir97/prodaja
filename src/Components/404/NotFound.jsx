import React from "react";
import NotFound from "../../assets/404.png";
import Style from "./style";
import { Typography, Grid, Button } from "@material-ui/core";
export default ({ history }) => {
  return (
    <Grid container direction="column" alignItems="center" style={Style.root}>
      <img style={Style.img} src={NotFound} alt="404" />
      <Typography variant="h5" style={Style.notFound}>
        Opps... Page not found
      </Typography>
      <Button style={Style.button} onClick={() => history.push("/")}>
        <Typography variant="h5" style={Style.text}>
          go back to Home
        </Typography>
      </Button>
    </Grid>
  );
};
