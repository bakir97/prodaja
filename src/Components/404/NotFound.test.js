import NotFound from "./index";
describe("NotFound", () => {
  const history = {
    push: jest.fn()
  };
  const wrapper = shallow(<NotFound history={history} />);
  it("should render normal", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call history", () => {
    wrapper.find("#button").simulate("click");
    expect(history.push).toBeCalledWith("/");
  });
});
