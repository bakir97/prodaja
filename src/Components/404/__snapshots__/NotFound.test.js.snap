// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render normal 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  direction="column"
  style={
    Object {
      "height": "calc(100vh - 7rem)",
      "marginTop": "7rem",
    }
  }
>
  <img
    alt="404"
    src="404.png"
    style={
      Object {
        "height": 300,
        "maxWidth": 500,
        "width": "100%",
      }
    }
  />
  <WithStyles(Typography)
    style={
      Object {
        "fontFamily": "Kalam",
        "marginTop": "2rem",
      }
    }
    variant="h5"
  >
    Opps... Page not found
  </WithStyles(Typography)>
  <WithStyles(Button)
    id="button"
    onClick={[Function]}
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    <WithStyles(Typography)
      style={
        Object {
          "WebkitBackgroundClip": "text",
          "background": "linear-gradient(to right, #373b44, #4286f4)",
          "color": "transparent",
          "fontFamily": "Kalam",
          "paddingBottom": 0,
        }
      }
      variant="h5"
    >
      go back to Home
    </WithStyles(Typography)>
  </WithStyles(Button)>
</WithStyles(Grid)>
`;
